activiti-core/activiti-api-impl/activiti-api-task-runtime-impl/src/main/java/org/activiti/runtime/api/model/impl/APITaskConverter.java

```diff
@@ -75,6 +74,15 @@ public Task from(org.activiti.engine.task.Task internalTask,
         return task;
     }
 
+    public Task fromWithCompletedBy(org.activiti.engine.task.Task internalTask,
+        Task.TaskStatus status, String completedBy) {
+
+        TaskImpl task = (TaskImpl) from(internalTask, status);
```

Comment:
I'm wondering if we cannot avoid this cast. Maybe we can extract a private method `buildFromInternalTask` that takes `org.activiti.engine.task.Task internalTask, Task.TaskStatus status` as parameters and returns `TaskImpl` instead of `Task`. This would avoid casting here and on `fromWithCandidates`. What do you think?